localstack/services/providers.py

```diff
@@ -319,19 +319,14 @@ def events():
 
 @aws_provider()
 def stepfunctions():
-    from localstack.services.stepfunctions.provider import StepFunctionsProvider
+    from localstack.services.stepfunctions.provider_v2 import StepFunctionsProvider
 
     provider = StepFunctionsProvider()
-    return Service.for_provider(
-        provider,
-        dispatch_table_factory=lambda _provider: HttpFallbackDispatcher(
-            _provider, _provider.get_forward_url
-        ),
-    )
+    return Service.for_provider(provider)
 
 
-@aws_provider(api="stepfunctions", name="v1")
-def stepfunctions_v1():
+@aws_provider(api="stepfunctions", name="legacy")
```

Comment:
Maybe we should keep the `v1` and `v2` aliases here in order to avoid any breaking changes for users who explicitly selected one or another? In addition we could maybe add a deprecation notice to make users aware of the fact that they should remove the `PROVIDER_OVERRIDE_STEPFUNCTIONS`?

Reply:
Both aliases have been added again. This is now following the same procedure we've been introducing with s3/lambda migrations.

The v1 provider already had a deprecation warning in its lifecycle hook. I've modified it to reflect the new situation. 

When we remove the v1 provider completely (4.0) we can add a corresponding entry to `deprecations.py` (same as we did with lambda).